<?xml version="1.0" encoding="UTF-8"?>
<testng-results retried="2" ignored="0" total="7" passed="4" failed="1" skipped="0">
  <reporter-output>
  </reporter-output>
  <suite started-at="2024-07-07T01:30:23 IST" name="Suite" finished-at="2024-07-07T01:31:16 IST" duration-ms="53737">
    <groups>
      <group name="ErrorHandling">
        <method signature="ErrorValidationTest.LogInErrorValidation()[pri:0, instance:RahulShettyAcademy.Tests.ErrorValidationTest@48075da3]" name="LogInErrorValidation" class="RahulShettyAcademy.Tests.ErrorValidationTest"/>
      </group> <!-- ErrorHandling -->
      <group name="purchase">
        <method signature="SubmitOrderTest.submitOrder(java.util.HashMap)[pri:0, instance:RahulShettyAcademy.Tests.SubmitOrderTest@7a6d7e92]" name="submitOrder" class="RahulShettyAcademy.Tests.SubmitOrderTest"/>
      </group> <!-- purchase -->
    </groups>
    <test started-at="2024-07-07T01:30:23 IST" name="SubmitOrder Test" finished-at="2024-07-07T01:31:13 IST" duration-ms="50467">
      <class name="RahulShettyAcademy.Tests.SubmitOrderTest">
        <test-method is-config="true" signature="launchApplication()[pri:0, instance:RahulShettyAcademy.Tests.SubmitOrderTest@7a6d7e92]" started-at="2024-07-07T01:30:23 IST" name="launchApplication" finished-at="2024-07-07T01:30:33 IST" duration-ms="9712" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- launchApplication -->
        <test-method signature="submitOrder(java.util.HashMap)[pri:0, instance:RahulShettyAcademy.Tests.SubmitOrderTest@7a6d7e92]" started-at="2024-07-07T01:30:33 IST" name="submitOrder" data-provider="getData" finished-at="2024-07-07T01:30:42 IST" duration-ms="8824" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[{password=Bhabani@123, product=ZARA COAT 3, email=Bhabani.sk.parida@gmail.com}]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- submitOrder -->
        <test-method is-config="true" signature="tearDown()[pri:0, instance:RahulShettyAcademy.Tests.SubmitOrderTest@7a6d7e92]" started-at="2024-07-07T01:30:42 IST" name="tearDown" finished-at="2024-07-07T01:30:45 IST" duration-ms="3376" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- tearDown -->
        <test-method is-config="true" signature="launchApplication()[pri:0, instance:RahulShettyAcademy.Tests.SubmitOrderTest@7a6d7e92]" started-at="2024-07-07T01:30:45 IST" name="launchApplication" finished-at="2024-07-07T01:30:54 IST" duration-ms="8844" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- launchApplication -->
        <test-method signature="submitOrder(java.util.HashMap)[pri:0, instance:RahulShettyAcademy.Tests.SubmitOrderTest@7a6d7e92]" started-at="2024-07-07T01:30:54 IST" name="submitOrder" data-provider="getData" finished-at="2024-07-07T01:30:59 IST" duration-ms="4716" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[{password=Prangya@123, product=IPHONE 13 PRO, email=kar.prangya123@gmail.com}]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- submitOrder -->
        <test-method is-config="true" signature="tearDown()[pri:0, instance:RahulShettyAcademy.Tests.SubmitOrderTest@7a6d7e92]" started-at="2024-07-07T01:30:59 IST" name="tearDown" finished-at="2024-07-07T01:31:00 IST" duration-ms="1117" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- tearDown -->
        <test-method is-config="true" signature="launchApplication()[pri:0, instance:RahulShettyAcademy.Tests.SubmitOrderTest@7a6d7e92]" started-at="2024-07-07T01:31:00 IST" name="launchApplication" finished-at="2024-07-07T01:31:10 IST" duration-ms="10401" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- launchApplication -->
        <test-method signature="orderHistory()[pri:0, instance:RahulShettyAcademy.Tests.SubmitOrderTest@7a6d7e92]" started-at="2024-07-07T01:31:10 IST" name="orderHistory" finished-at="2024-07-07T01:31:12 IST" depends-on-methods="RahulShettyAcademy.Tests.SubmitOrderTest.submitOrder" duration-ms="1588" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- orderHistory -->
        <test-method is-config="true" signature="tearDown()[pri:0, instance:RahulShettyAcademy.Tests.SubmitOrderTest@7a6d7e92]" started-at="2024-07-07T01:31:12 IST" name="tearDown" finished-at="2024-07-07T01:31:13 IST" duration-ms="1189" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- tearDown -->
      </class> <!-- RahulShettyAcademy.Tests.SubmitOrderTest -->
    </test> <!-- SubmitOrder Test -->
    <test started-at="2024-07-07T01:30:23 IST" name="ErrorValidation Test" finished-at="2024-07-07T01:31:16 IST" duration-ms="53737">
      <class name="RahulShettyAcademy.Tests.ErrorValidationTest">
        <test-method is-config="true" signature="launchApplication()[pri:0, instance:RahulShettyAcademy.Tests.ErrorValidationTest@48075da3]" started-at="2024-07-07T01:30:23 IST" name="launchApplication" finished-at="2024-07-07T01:30:36 IST" duration-ms="12703" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- launchApplication -->
        <test-method signature="LogInErrorValidation()[pri:0, instance:RahulShettyAcademy.Tests.ErrorValidationTest@48075da3]" started-at="2024-07-07T01:30:36 IST" name="LogInErrorValidation" finished-at="2024-07-07T01:30:37 IST" duration-ms="1846" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- LogInErrorValidation -->
        <test-method is-config="true" signature="tearDown()[pri:0, instance:RahulShettyAcademy.Tests.ErrorValidationTest@48075da3]" started-at="2024-07-07T01:30:37 IST" name="tearDown" finished-at="2024-07-07T01:30:38 IST" duration-ms="954" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- tearDown -->
        <test-method is-config="true" signature="launchApplication()[pri:0, instance:RahulShettyAcademy.Tests.ErrorValidationTest@48075da3]" started-at="2024-07-07T01:30:38 IST" name="launchApplication" finished-at="2024-07-07T01:30:49 IST" duration-ms="10959" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- launchApplication -->
        <test-method retried="true" signature="productErrorValidation()[pri:0, instance:RahulShettyAcademy.Tests.ErrorValidationTest@48075da3]" started-at="2024-07-07T01:30:49 IST" name="productErrorValidation" finished-at="2024-07-07T01:30:50 IST" duration-ms="1137" status="SKIP">
          <exception class="java.lang.NullPointerException">
            <full-stacktrace>
              <![CDATA[java.lang.NullPointerException
at RahulShettyAcademy.pageobjects.productCataloguePage.addProductTocart(productCataloguePage.java:49)
at RahulShettyAcademy.Tests.ErrorValidationTest.productErrorValidation(ErrorValidationTest.java:30)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:566)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:677)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:221)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:962)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:194)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
at org.testng.TestRunner.privateRun(TestRunner.java:806)
at org.testng.TestRunner.run(TestRunner.java:601)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:433)
at org.testng.SuiteRunner$SuiteWorker.run(SuiteRunner.java:471)
at org.testng.internal.thread.ThreadUtil.lambda$execute$0(ThreadUtil.java:58)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
at java.base/java.lang.Thread.run(Thread.java:834)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.NullPointerException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- productErrorValidation -->
        <test-method is-config="true" signature="tearDown()[pri:0, instance:RahulShettyAcademy.Tests.ErrorValidationTest@48075da3]" started-at="2024-07-07T01:30:50 IST" name="tearDown" finished-at="2024-07-07T01:30:52 IST" duration-ms="1178" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- tearDown -->
        <test-method is-config="true" signature="launchApplication()[pri:0, instance:RahulShettyAcademy.Tests.ErrorValidationTest@48075da3]" started-at="2024-07-07T01:30:52 IST" name="launchApplication" finished-at="2024-07-07T01:31:01 IST" duration-ms="9487" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- launchApplication -->
        <test-method retried="true" signature="productErrorValidation()[pri:0, instance:RahulShettyAcademy.Tests.ErrorValidationTest@48075da3]" started-at="2024-07-07T01:31:01 IST" name="productErrorValidation" finished-at="2024-07-07T01:31:03 IST" duration-ms="1938" status="SKIP">
          <exception class="java.lang.NullPointerException">
            <full-stacktrace>
              <![CDATA[java.lang.NullPointerException
at RahulShettyAcademy.pageobjects.productCataloguePage.addProductTocart(productCataloguePage.java:49)
at RahulShettyAcademy.Tests.ErrorValidationTest.productErrorValidation(ErrorValidationTest.java:30)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:566)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:677)
at org.testng.internal.invokers.TestInvoker.retryFailed(TestInvoker.java:263)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:62)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:962)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:194)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
at org.testng.TestRunner.privateRun(TestRunner.java:806)
at org.testng.TestRunner.run(TestRunner.java:601)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:433)
at org.testng.SuiteRunner$SuiteWorker.run(SuiteRunner.java:471)
at org.testng.internal.thread.ThreadUtil.lambda$execute$0(ThreadUtil.java:58)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
at java.base/java.lang.Thread.run(Thread.java:834)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.NullPointerException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- productErrorValidation -->
        <test-method is-config="true" signature="tearDown()[pri:0, instance:RahulShettyAcademy.Tests.ErrorValidationTest@48075da3]" started-at="2024-07-07T01:31:03 IST" name="tearDown" finished-at="2024-07-07T01:31:04 IST" duration-ms="1036" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- tearDown -->
        <test-method is-config="true" signature="launchApplication()[pri:0, instance:RahulShettyAcademy.Tests.ErrorValidationTest@48075da3]" started-at="2024-07-07T01:31:04 IST" name="launchApplication" finished-at="2024-07-07T01:31:15 IST" duration-ms="10979" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- launchApplication -->
        <test-method signature="productErrorValidation()[pri:0, instance:RahulShettyAcademy.Tests.ErrorValidationTest@48075da3]" started-at="2024-07-07T01:31:15 IST" name="productErrorValidation" finished-at="2024-07-07T01:31:16 IST" duration-ms="1038" status="FAIL">
          <exception class="java.lang.NullPointerException">
            <full-stacktrace>
              <![CDATA[java.lang.NullPointerException
at RahulShettyAcademy.pageobjects.productCataloguePage.addProductTocart(productCataloguePage.java:49)
at RahulShettyAcademy.Tests.ErrorValidationTest.productErrorValidation(ErrorValidationTest.java:30)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:566)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:677)
at org.testng.internal.invokers.TestInvoker.retryFailed(TestInvoker.java:263)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:62)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:962)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:194)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
at org.testng.TestRunner.privateRun(TestRunner.java:806)
at org.testng.TestRunner.run(TestRunner.java:601)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:433)
at org.testng.SuiteRunner$SuiteWorker.run(SuiteRunner.java:471)
at org.testng.internal.thread.ThreadUtil.lambda$execute$0(ThreadUtil.java:58)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
at java.base/java.lang.Thread.run(Thread.java:834)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.NullPointerException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- productErrorValidation -->
        <test-method is-config="true" signature="tearDown()[pri:0, instance:RahulShettyAcademy.Tests.ErrorValidationTest@48075da3]" started-at="2024-07-07T01:31:16 IST" name="tearDown" finished-at="2024-07-07T01:31:16 IST" duration-ms="320" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- tearDown -->
      </class> <!-- RahulShettyAcademy.Tests.ErrorValidationTest -->
    </test> <!-- ErrorValidation Test -->
  </suite> <!-- Suite -->
</testng-results>
